scipy/stats/_qmc.py

```diff
@@ -503,25 +505,24 @@ def van_der_corput(
        arXiv:1706.02808, 2017.
 
     """
-    rng = check_random_state(seed)
-    sequence = np.zeros(n)
+    if base < 2:
+        raise ValueError('base must be at least 2')
+
+    if scramble:
+        rng = check_random_state(seed)
+        # Create a set of permutation of np.arange(base) for each positive
+        # integer k such that base**-k > 2**-54. The number of permutations is
+        # equal to the number of iterations in the while loop in Algorithm 1 of
+        # Owen 2017.
+        count = math.ceil(54 / math.log2(base)) - 1
+        permutations = np.repeat(np.arange(base)[None], count, axis=0)
+        for perm in permutations:
+            rng.shuffle(perm)
```

Comment:
> Where does 54 comes from?

54 is the number of bits in the mantissa for a double precision float plus 2. The condition in the while loop in Algorithm 1, `1 - base**-k < 1`, can also be written as `base**-k > 2**-(number_of_mantissa_bits + 2)`, which makes it more clear how many iterations are in the while loop (i.e., number of positive integers `k` such that `base**-k > 2**-(number_of_mantissa_bits + 2)`). I will update the comment to try to make this more clear.

> Any reason why this is not in Cython?

The permutations could be generated in `_cy_van_der_corput_scrambled`, but there would not be any benefit, since the code generating the random permutations would still be running at the python-level. 

> I am wondering if this is correct to do the permutations here. I had some experiments where I was doing the permutations not in the loop and had worse performances. I might be missing something here.

It should not matter.  I have been comparing the results of the cythonized code to this less efficient implementation that closely follows Algorithm 1:

```python
import numpy as np                                                                                                                                                                                          
                                                                                                                                                                                                            
def van_der_corput_owen(n, b, rng):                                                                                                                                                                         
    ind = np.arange(n)                                                                                                                                                                                      
    b2r = 1/b                                                                                                                                                                                               
    ans = ind*0.0                                                                                                                                                                                           
    res = ind                                                                                                                                                                                               
    while (1 - b2r) < 1:                                                                                                                                                                                    
        dig = res % b                                                                                                                                                                                       
        pi = rng.permutation(b)                                                                                                                                                                             
        pdig = pi[dig]                                                                                                                                                                                      
        ans = ans + pdig*b2r                                                                                                                                                                                
        b2r = b2r/b                                                                                                                                                                                         
        res = (res - dig)/b                                                                                                                                                                                 
        res = res.astype(int)                                                                                                                                                                               
                                                                                                                                                                                                            
    return ans 
```
The results should be identical.